### ISSUE TITLE
Stop breaking ethereum.js

### ISSUE THREAD

**Issue by caktux** [ethereum/ethereum.js#113]
How hard is it to just improve things at every commit? If there's one part of the stack we expect to **just work**, it's ethereum.js. The deprecation warnings were a good example of that despite the new getBalance not behaving like the older balanceAt which actually was still working unlike the new method.
So please, stop breaking ethereum.js, and do a little more testing before every commit, not just running the included tests. Think a little further than you're used to, because other people and other projects depend on this critical piece of software, and there's going to be more and more of those soon enough.
Please take this seriously, this is more than just a rant. Thank you.

**Comment by chevdor** [comment]
had a bad day :-p ?

**Comment by kumavis** [comment]
develop branch is unstable by definition -- are you running into problems on master?

**Comment by caktux** [comment]
Unstable by definition does not justify broken and unusable. Develop fails with #111, master fails with #112, and the npm module is simply too outdated. Unstable means a method or feature might not work as expected or will fail, but the project as a whole should always be installable and usable at a minimum. **Use features branches for breaking the whole thing, not the main branches.** Take cpp-ethereum for example, its develop branch is highly unstable, but very rarely will it ever be broken the way ethereum.js is half the time. There is a clear lack of thoroughness and attention to detail in this project which needs to be addressed, and this is what I'm pointing out. Do some real integration tests before pushing to develop, not just the npm ones. Install it as an npm module also or write tests for that. Document the disparities as they occur between the API and the ethereum clients. Pay attention to details, that is the point here.
Just to be clear, I'm not directing this at you, on the contrary. You've been doing a great job on this project and you provided very important fixes, some of which are still pending for some reason.

**Comment by debris** [comment]
That's true, recently I have merged a lot of braking changes into master branch. We tried to mark every possible method as deprecated, but unfortunately we skipped few things. In future we will try to pay more attention to the details.
On the bright side, we do not plan to do any more breaking changes in ethereum.js or JSON-RPC. From now on, we will focus on cleaning the code && writing more tests.
You are doing great job @catux and also pointing out these problems is needed. I will try to address the issues that you reported asap.
